src/components/core/CardList.tsx

```diff
@@ -0,0 +1,40 @@
+import { ReactElement, useCallback } from 'react'
+
+import {
+  FlatList,
+  FlatListProps,
+  ListRenderItemInfo,
+  ViewStyle,
+  StyleProp,
+  ListRenderItem
+} from 'react-native'
+
+type ListProps<ItemT> = Omit<FlatListProps<ItemT>, 'renderItem'>
+
+type CardListRenderItemInfo<ItemT> = ListRenderItemInfo<ItemT> & {
+  style: StyleProp<ViewStyle>
+}
+
+export type CardListProps<ItemT> = ListProps<ItemT> & {
+  renderItem: (info: CardListRenderItemInfo<ItemT>) => ReactElement | null
+}
+
+export function CardList<ItemT>(props: CardListProps<ItemT>) {
```

Comment:
Did you make this a named function to allow for the type parameter? Another option (not necessarily better) I've discovered is:

```suggestion
export const CardList = <ItemT, _>(props: CardListProps<ItemT>) => {
```

The unused type param lets JSX figure out that it's not a component

Reply:
totally, its just like... ugh why ts... :/ can update if that's the standard!

Reply:
It's not a standard! both ways are annoying for sure